odrive-motor/src/odriveCAN.py

```diff
@@ -0,0 +1,290 @@
+        msg = can.Message(arbitration_id=0x00E | self.nodeID << 5, is_extended_id=False, data=data)
+        await self.try_send(msg)
+
+    async def go_for(self, rpm: float, revolutions: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        rps = rpm / MINUTE_TO_SECOND
+
+        msg = db.get_message_by_name('Set_Axis_State')
+        data = msg.encode({'Axis_Requested_State': 0x08})
+        msg = can.Message(arbitration_id=msg.frame_id | self.nodeID << 5, is_extended_id=False, data=data)
+        await self.try_send(msg)
+        await self.wait_until_correct_state(AxisState.CLOSED_LOOP_CONTROL)
+
+        if revolutions == 0.0:
+            msg = db.get_message_by_name('Set_Controller_Mode')
+            data = msg.encode({'Control_Mode': 0x02, 'Input_Mode': 0x01})
+            msg = can.Message(arbitration_id=0x00B | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+            
+            msg = db.get_message_by_name('Set_Input_Vel')
+            data = msg.encode({'Input_Vel': rps, 'Input_Torque_FF': 0})
+            msg = can.Message(arbitration_id=0x00D | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+
+        else: 
+            msg = db.get_message_by_name('Set_Controller_Mode')
+            data = msg.encode({'Control_Mode': 0x03, 'Input_Mode': 0x05})
+            msg = can.Message(arbitration_id=0x00B | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+
+            msg = db.get_message_by_name('Set_Limits')
+            data = msg.encode({'Velocity_Limit': 10.0, 'Current_Limit': 40.0})
+            msg = can.Message(arbitration_id=0x00F | self.nodeID << 5, is_extended_id=False, data=data)
+            await self.try_send(msg)
+
+            current_position = await self.get_position()
+            if rpm > 0.0:
+                msg = db.get_message_by_name('Set_Input_Pos')
+                goal_position = current_position+revolutions+self.offset
+                data = msg.encode({'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
+                msg = can.Message(arbitration_id=0x00C | self.nodeID << 5, is_extended_id=False, data=data)
+                await self.try_send(msg)
+            else:
+                msg = db.get_message_by_name('Set_Input_Pos')
+                goal_position = current_position-revolutions+self.offset
+                data = msg.encode({'Input_Pos': (goal_position), 'Vel_FF': 0, 'Torque_FF': 0})
+                msg = can.Message(arbitration_id=0x00C | self.nodeID << 5, is_extended_id=False, data=data)
+                await self.try_send(msg)
+
+            await self.wait_and_set_to_idle(goal_position)
+
+    async def go_to(self, rpm: float, revolutions: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        current_position = await self.get_position()
+        revolutions = revolutions - current_position
```

Comment:
handle case where this results in a velocity command sent even if the position difference is close to zero.  Do this in the serial driver as well.
If my GoTo passes 10 and I'm already at position 10, it'll start moving at the input rpm forever.